### PR TITLE
LOK-1899: Add Spring Security to BFF

### PR DESCRIPTION
**Pull request by vserrago**
## Description
Adds spring security to the GraphQL BFF. Previously, the user token was only verfied/authorized when a grpc client was attempting to make a call. This would result in things like the request returning a graphQL datafetch error because the user's token was expired. Now instead, it is done via a request filter chain via Spring Security.

Right now we don't have much in the way of RBAC, but we are more primed now to use it via further customization of `grantedAuthoritiesExtractor`.

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-1899

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

**Review by bouff — APPROVED** [review]
LGTM!

Nice and tidy implementation 🚀

**Review by freddy61025 — APPROVED** [review]
LGTM just some minor comments

File: rest-server/src/main/resources/application.yml [hunk 1]
```diff
@@ -9,11 +9,14 @@ server:
...
-  url: http://onms-keycloak:8080/auth
-  realm: opennms
-  signatureAlgorithm: RS256
+  security:
+    oauth2:
+      resourceserver:
+        jwt:
+          # Providing this value will cause Spring to autoconfigure a
+          # org.springframework.security.oauth2.jwt.ReactiveJwtDecoder instance.
+          jwk-set-uri: http://onms-keycloak:8080/auth/realms/opennms/protocol/openid-connect/certs
```
[the remark this answers is not in the document]
Comment (vserrago) [reply rest-server/src/main/resources/application.yml]: We don't have a helm chart configuration for the previous values; we simply use the hardcoded values in `application.yml` everywhere.

However on talking with DJ about it, we'll probably want to expose those as configurable in the helm charts across all services. That will be done in a ticket that I will create soon. For this PR, I have re-added these properties and redefined the `jwk-set-*` fields to reference them instead.

**Review by freddy61025 — APPROVED** [review]
LGTM

**Pull request merged**